src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -269,13 +321,109 @@ public void DeleteIndex([NotNull] string name)
 
         #endregion
 
-        #region IDispose
+        #region Public Methods - QueryFactory
 
-        public void Dispose()
+        public IQuery CreateQuery([NotNull] string queryExpression)
         {
             throw new NotImplementedException();
         }
 
         #endregion
+
+        #region Internal Methods
+
+        internal void GetSpec()
+        {
+            ThreadSafety.DoLocked(() =>
+            {
+                var spec = Native.c4coll_getSpec(c4coll);
+                Name = spec.name.CreateString();
+                Scope.Name = spec.scope.CreateString();
+            });
+        }
+
+        /// <summary>
+        /// Returns false if this collection has been deleted, or its database closed.
+        /// </summary>
+        internal void CheckCollectionValid()
+        {
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4Db == null) {
+                    throw new InvalidOperationException(CouchbaseLiteErrorMessage.DBClosed);
+                }
+
+                var isValid = Native.c4coll_isValid((C4Collection*)_c4coll);
+                if (!isValid || _c4coll == IntPtr.Zero) {
+                    throw new InvalidOperationException(String.Format(CouchbaseLiteErrorMessage.CollectionNotAvailable,
+                                ToString()));
+                }
+            });
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private C4Database* GetC4Database()
+        {
+            C4Database* c4db = null;
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4coll == null)
+                    return;
+
+                c4db = Native.c4coll_getDatabase(c4coll);
+            });
+
+            return c4db;
+        }
+
+        private unsafe void ReleaseCollection()
+        {
+            IntPtr temp = IntPtr.Zero;
+            Interlocked.Exchange(ref temp, _c4coll);
```

Comment:
This doesn't actually help anything, as I mentioned, so either _c4coll needs to be an IntPtr, or just live with a simple nonatomic swap and release.